e2e/test/scenarios/native-filters/helpers/e2e-date-filter-helpers.js

```diff
@@ -24,6 +24,14 @@ export function setSingleDate(date) {
   setDate(date, cy.findByTestId("specific-date-picker"));
 }
 
+export function setTime({ hh: hours, mm: minutes }) {
+  popover().within(() => {
+    cy.findByText("Add a time").click();
```

Comment:
I suggest to explicitly assert here so that this function is not accidentally removed in the future if it starts giving errors:

```suggestion
    const addTime = cy.findByText("Add a time");
    expect(addTime).toBeInTheDocument();
    addTime.click();
```

Reply:
actually `findByText` will fail if it doesn't find such text. also it's e2e test 😄

Reply:
> actually `findByText` will fail if it doesn't find such text.

I'm aware, I'm suggesting to make an "explicit" assertion here. Otherwise it looks like a helper function.
If a helper function does not work as I expect I start looking for issues in test setup.
If an asserting function does not work as I expect, I start looking for issues in the actual code.

> also it's e2e test :smile:

Please treat all code suggestions as pseudocode :smile:

Reply:
let me summon @nemanjaglumac to the thread. Do you think we need to explicitly assert elements in cypress tests?

`cy.findByText(selector).click();` - do you expect this will fail if `selector` is not in the document?

Reply:
> Do you think we need to explicitly assert elements in cypress tests?

Just to be clear: there is nothing technical about my suggestion to assert this explicitly. My suggestion is purely about code readability.

Reply:
Or alternative suggestion: let's rename `setTime` to `assertCanSetTime`.

Reply:
okay, in this case let's consider this is just a helper to set time. actual assert happens on the test and checks if the resulted data shown on the page is correct

Reply:
Without going into details of whether or not we need to be explicit here, I do want to point to a stylistic thing.

This will not work to the best of my knowledge. You can use `.should` and assert the element is visible.
Btw, click fails if an element:
- doesn't exist
- is not visible
- is not "actionable"

```diff
-   const addTime = cy.findByText("Add a time");
-   expect(addTime).toBeInTheDocument();
-   addTime.click();
+ cy.findByText("Add a time").should("be.visible").click();
```

And to answer the question of whether or not we need to make explicit assertions. I don't think so.

> do you expect this will fail if `selector` is not in the document?

Yep, I do.
I also don't think that anyone should go and remove a piece of code (be it helper or inline code) if test starts failing. If someone does that, we have a bigger problem to deal with.